frontend/src/components/pages/RegistrantExperience/Waiver/index.tsx

```diff
@@ -1,13 +1,175 @@
+          above.
+        </Text>
+        <Checkbox
+          isChecked={waiverInterface.agreedRequiredClauses}
+          onChange={() =>
+            waiverDispatch({ type: WaiverActions.ClICK_REQUIRED_CLAUSE })
+          }
+        >
+          <Text textStyle="bodyRegular">
+            I agree to the above identity and liability waiver conditions{" "}
+            <Text as="span" color="text.critical.100" verticalAlign="super">
+              *
+            </Text>
+          </Text>
+        </Checkbox>
+      </VStack>
+      <Divider pt="4" pb="4" borderColor="border.secondary.100" />
+      <VStack spacing={5} pt="8" align="stretch">
+        <Text color="primary.green.100" textStyle="heading">
+          Additional Clauses
+        </Text>
+        {waiverInterface.optionalClauses.map(
+          (clause: OptionalClauseResponse, index: number) => (
+            <VStack spacing={6} pt="2" align="stretch" key={index}>
+              <Text textStyle="bodyRegular">{clause.text}</Text>
+              <RadioGroup
+                onChange={() =>
+                  waiverDispatch({
+                    type: WaiverActions.CLICK_OPTIONAL_CLAUSE,
+                    optionalClauseId: index,
+                  })
+                }
+                key={index}
+                value={String(clause.agreed)}
+              >
+                <Stack spacing={5} direction="row">
+                  <Radio value="true" mb="0" textStyle="bodyRegular">
+                    {/* NOTE!?: I use mb='0' because there's a margin misalignment issue with Radio component */}
+                    I agree
+                  </Radio>
+                  <Radio value="false" textStyle="bodyRegular">
+                    I{" "}
+                    <Text as="span" textStyle="bodyBold">
+                      do not
+                    </Text>{" "}
+                    agree
+                  </Radio>
+                </Stack>
+              </RadioGroup>
+            </VStack>
+          ),
+        )}
+      </VStack>
+      <Divider
+        pt="4"
+        pb="4"
+        borderColor="border.secondary.100"
+        textStyle="heading"
+      />
+      <Text textStyle="heading" pt="6">
+        By typing my name below, I hereby declare that I fully release and
+        discharge Focus on Nature, its employees and volunteers from all
+        liabilities to which I have agreed to above.
+      </Text>
+      <HStack pt="4" spacing="6">
+        <FormControl width="30%">
+          <FormLabel>
+            <Text textStyle="buttonSemiBold">
+              Full Name{" "}
+              <Text
+                as="span"
+                color="text.critical.100"
+                fontSize="xs"
+                verticalAlign="super"
+              >
+                *
+              </Text>
+            </Text>
+          </FormLabel>
+          <Input textStyle="bodyRegular" fontFamily="cursive" />
+        </FormControl>
```

Comment:
I think we need to bind this to a state variable in the parent component to make sure the form is filled out, and only allow the user to proceed to the next page if they've accepted everything. Not sure if we want to scope that into this PR or keep it as a separate PR?

Another option is using Chakra's built-in `isInvalid` in `<FormControl>`, but it often struggles with the zero state empty (we don't really want the initial state of the form to be an error state)

Reply:
Yeah let me do it. I'll just make a boolean state for whether the name and date are filled. Should just be a small change.